### PR TITLE
Add an index defined type

### PR DESCRIPTION
**Pull request by npwalker**

### CODE REVIEW

File: manifests/server/index.pp [hunk 1]
```diff
@@ -0,0 +1,35 @@
+define postgresql::server::index (
+  Enum['absent','present'] $ensure,
+  String        $db,
+  String        $index_name,
```
Comment (hasegeli) [inline manifests/server/index.pp]: We can default this to `$name`.

File: manifests/server/index.pp [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  String        $table_name,
+  Array[String] $column_names,
```
Comment (hasegeli) [inline manifests/server/index.pp]: It better be `Array[String[1], 1]`.

File: manifests/server/index.pp [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  String        $psql_user         = $postgresql::server::user,
+  String        $psql_group        = $postgresql::server::group,
+  Integer       $port              = $postgresql::server::port,
+  Hash          $connect_settings  = $postgresql::server::default_connect_settings,
+  String        $psql_path         = $postgresql::server::psql_path
```
Comment (hasegeli) [inline manifests/server/index.pp]: `concurrently`, `unique`, `index_type`, `predicate` options would be nice.

https://www.postgresql.org/docs/9.5/static/sql-createindex.html

File: manifests/server/index.pp (hunk 2 above)
Comment (hasegeli) [inline manifests/server/index.pp]: `_name`s are redundant and not used on others.

**Comment by hasegeli** [comment]
Thank you for submitting this.  I think we should also maintain the README.

File: manifests/server/index.pp [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    default => '',
+  }
+
+  $_concurrently = $concurrently ? {
+    true    => 'CONCURRENTLY',
+    default => '',
+  }
+
+  $with_predicate = empty($predicate) ? {
+    false   => "WITH ${predicate}"
```
Comment (hasegeli) [inline manifests/server/index.pp]: `WHERE`

**Comment by david22swan** [comment]
@npwalker Is there any movement on this pr?

**Comment by davidmalloncares** [comment]
Hi @npwalker we are going to close this PR in order to progress with planned work we have in flight. If you want to rebase and raise a new PR with these changes along with the appropriate tests, readme, etc we can reassess then.

**Pull request closed without merging**